packages/discord.js/src/managers/GuildBanManager.js

```diff
@@ -199,6 +199,42 @@ class GuildBanManager extends CachedManager {
     await this.client.rest.delete(Routes.guildBan(this.guild.id, id), { reason });
     return this.client.users.resolve(user);
   }
+
+  /**
+   * Bulk ban users from a guild, and optionally delete previous messages sent by the banned users.
+   * @param {Collection<Snowflake, User|UserResolvable>|UserResolvable[]} users The users to ban
+   * @param {BanOptions} [options] The options for bulk banning users
+   * @returns {Promise<BanBulkResult>} Returns an object with `bannedUsers` key containing the IDs of the banned users
+   * and the key `failedUsers` with the IDs that could not be banned or were already banned.
+   * @example
+   * // Bulk ban users by ids (or with user/guild member objects)
+   * guild.bans.bulkCreate(['84484653687267328'], { deleteMessageSeconds: 7 * 24 * 60 * 60 })
+   *   .then(result => {
+   *     console.log(`Banned ${result.bannedUsers.length} users, failed to ban ${result.failedUsers.length} users.`)
+   *   })
+   *   .catch(console.error);
+   */
+  async bulkCreate(users, options = {}) {
+    if (!users) throw new DiscordjsError(ErrorCodes.BulkBanUsersOptionRequired);
+    if (typeof options !== 'object') throw new DiscordjsTypeError(ErrorCodes.InvalidType, 'options', 'object', true);
+
+    if (!Array.isArray(users) && !(users instanceof Collection)) {
+      throw new DiscordjsTypeError(ErrorCodes.BulkBanUsersOptionInvalid, typeof users, 'array or collection');
+    }
+
+    const userIds = users.map(user => this.client.users.resolveId(user));
+    if (userIds.length === 0) throw new DiscordjsError(ErrorCodes.BulkBanUsersOptionEmpty);
+
+    const result = await this.client.rest.post(Routes.guildBulkBan(this.guild.id), {
+      body: {
+        delete_message_seconds:
+          options.deleteMessageSeconds ??
+          (options.deleteMessageDays ? options.deleteMessageDays * 24 * 60 * 60 : undefined),
```

Comment:
IMO there's no reason to add support for a deprecated option (`deleteMessageDays`) to a new method